### PR TITLE
PHP8 compatibility

### PR DESCRIPTION
**Pull request by cadic**
### Description of the Change

Added tests for compatibility against PHP7 and PHP8:
- PHPCompatibilityWP 7.0, 7.4, 8.0, 8.1
- PHPUnit against 7.4, 8.1

Updated PHPUnit to 9.5, updated unit tests to be compatible with 9.5

### Verification Process

Check GitHub Actions workflows all pass.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Changelog Entry

Added - Compatibility tests against PHP 7 and 8.
Changed - Unit tests against PHP 8
Changed - Bump required PHP 7.0

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ Contributors: 10up, helen, adamsilverstein, jakemgold
 Tags: simple podcasting, podcasting, podcast, apple podcasts, episode, gutenberg, blocks, block
 Requires at least: 4.6
 Tested up to: 5.9
-Requires PHP: 5.3
+Requires PHP: 7.3
```
Comment (dkotter) [inline readme.txt]: Curious why we're bumping from 5.3 all the way to 7.3? Doesn't seem like we have any new features we're adding that require 7.3 so just wondering on the reason here? I think for sure we should drop support for 5.3 at this point but maybe we just bump to 7.0? We also need to update the README.md file as well, as that mentions 5.3 as the minimum supported version

**Comment by dkotter** [comment]
There's some merge conflicts here that need fixing

File: readme.txt (hunk 1 above)
Comment (cadic) [reply readme.txt]: I agree, we should keep 7.0 support. Updated both readme files and added 7.0 to compatibility testing.

The initial bump 7.3 was made together with composer dependency (PHPUnit 9.5 requires at least PHP 7.3). This is dev-only requirement and of course the plugin itself works well with 7.0.

**Comment by cadic** [comment]
@dkotter resolved conflicts and fixed PHP minimum reqs in readme files.

I also reduced the number of Unit tests just to cover latest PHP 7 and 8, not sure it makes sense to test against every release within a major version. We still test compatibility against 7.0, 7.4, 8.0 and 8.1 in separate workflow

**Review by dkotter — APPROVED** [review]

**Pull request merged**